stdlib/public/Distributed/DistributedActor.swift

```diff
@@ -140,8 +140,18 @@ extension DistributedActor {
   /// state.
   ///
   /// When the actor is remote, the closure won't be executed and this function will return nil.
+//  public nonisolated func whenLocal<T: Sendable>(
```

Comment:
I assume you want to remove this?

Reply:
I have to figure out how to introduce the "good ones" without breaking existing code; this one has unfortunate semantics that it does "await x.whenLocal <HOP HERE> { isolated X in .... }` whereas the other one this `_local` enables allows us to not hop unless we have to: `x.whenLocal { _local X in xx.someConstant }` so there's no hop and no async there...

We also want to allow `await x.whenLocal { _local X in await xx.something }` though; so we can either do this by keeping the old version which does the HOP once (could be good?), or making some new overload.

So in other words... perhaps keeping the "old" signature is good because:
- won't break API/ABI
- is could be okey...? we do one hop when the function is async and run on the actor...?

If we could get away with removing this function though I think we would want to -- can we though?

Reply:
Open question what to do here, the "I'd love those" signatures are:


```swift
  public nonisolated func whenLocal<T: Sendable>( // FIXME: do we need sendable req here?
    _ body: @Sendable (_local Self) async throws -> T
  ) async rethrows -> T? {
    if __isLocalActor(self) {
       return try await body(self)
    } else {
      return nil
    }
  }

  public nonisolated func whenLocal<T: Sendable>(
    _ body: (_local Self) throws -> T
  ) rethrows -> T? {
    if __isLocalActor(self) {
       return try body(self)
    } else {
      return nil
    }
  }
```


so we can avoid the "one hop" at the beginning, but I guess we COULD keep the isolated one as well hmm

Reply:
These signatures look fine to me, but I don't think you need the Sendable constraints because you're never actually crossing the actor boundary. But I think you do need to decide to either have the `_local` versions, or the `isolated` versions, but if you want both you're going to need to give them different names.